File: 백준/Silver/13567. 로봇/로봇.py

M, n = map(int, input().split())
commands = []
for _ in range(n):
    command, param = input().split()
    if command == "TURN":
        commands.append((0, int(param)))
    else:
        commands.append((1, int(param)))
valid_command_set = True
x = 0
y = 0
direction = 0
dirs = ((0, 1), (-1, 0), (0, -1), (1, 0))
for command, param in commands:
    if command:
        y += param * dirs[direction][0]
        x += param * dirs[direction][1]
    elif param == 1:
        direction = (direction + 1) % 4
    else:
        direction = (direction - 1) % 4
    if not (0 <= x < M and 0 <= y < M):
        valid_command_set = False
        break

if valid_command_set:
    print(x, y)
else:
    print(-1)

File: 백준/Silver/13567. 로봇/로봇.cc
#include <iostream>
using namespace std;

int M, n;
int y, x, dir;
pair<int, int> dirs[4] = {{0, 1}, {-1, 0}, {0, -1}, {1, 0}};

int main(void) {
    cin >> M >> n;
    y = 0;
    x = 0;
    dir = 0;
    bool valid_command_set = true;
    for (int i = 0; i < n; i++) {
        string command;
        int param;
        cin >> command >> param;
        if (command == "MOVE") {
            y += param * dirs[dir].first;
            x += param * dirs[dir].second;
        } else if (param == 1) {
            dir = (dir + 1) % 4;
        } else {
            dir = (dir + 3) % 4;
        }
        if (y < 0 || x < 0 || y >= M || x >= M) {
            valid_command_set = false;
            break;
        }
    }
    if (valid_command_set) printf("%d %d\n", x, y);
    else printf("-1\n");
}